presto-main/src/main/java/com/facebook/presto/sql/planner/ExpressionSymbolInliner.java

```diff
@@ -17,26 +17,41 @@
 import com.facebook.presto.sql.tree.ExpressionRewriter;
 import com.facebook.presto.sql.tree.ExpressionTreeRewriter;
 import com.facebook.presto.sql.tree.SymbolReference;
+import com.google.common.collect.ImmutableMap;
 
 import java.util.Map;
 
-import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Objects.requireNonNull;
 
 public class ExpressionSymbolInliner
         extends ExpressionRewriter<Void>
 {
-    private final Map<Symbol, ? extends Expression> mappings;
+    private final Map<Symbol, ? extends Expression> expressionMappings;
+    private final Map<Symbol, Symbol> symbolMappings;
```

Comment:
Instead of changing this to work on symbol->symbol, update the user to create a symbol->expression map.